packages/babel-plugin-proposal-decorators/CONTRIB.md

```diff
@@ -0,0 +1,374 @@
+These are notes about the implementation of the 2021-12 decorators transform.
+The implementation's goals are (in descending order):
+
+1. Being accurate to the actual proposal (e.g. not defining additional
+  properties unless required, matching semantics exactly, etc.). This includes
+  being able to work properly with private fields and methods.
+2. Transpiling to a very minimal and minifiable output. This transform will
+  affect each and every decorated class, so ensuring that the output is not 10x
+  the size of the original is important.
+3. Having good runtime performance. Decoration output has the potential to
+  drastically impact startup performance, since it runs whenever a decorated
+  class is defined. In addition, every instance of a decorated class may be
+  impacted for certain types of decorators.
+
+All of these goals come somewhat at the expense of readability and can make the
+implementation difficult to understand, so these notes are meant to document the
+motivations behind the design.
+
+## Overview
+
+Given a simple decorated class like this one:
+
+```js
+@dec
+class Class {
+  @dec a = 123;
+
+  @dec static #b() {
+    console.log('foo');
+  }
+
+  [someVal]() {}
+
+  @dec
+  @dec2
+  accessor #c = 456;
+}
+```
+
+It's output would be something like the following:
+
+```js
+import { applyDecs } from '@babel/helpers';
+
+let initInstance, initClass, initA, callB, computedKey, initC, getC, setC;
+
+const elementDecs = [
+  [dec, 0, 'a'],
+  [dec, 7, 'x', '#b']
```

Comment:
We need to update the example here because the method implementation is now moved to the static block so `x` is no longer required.